test/functorch/test_aot_joint_with_descriptors.py

```diff
@@ -761,6 +761,76 @@ def forward(self, x):
         compiled_fn(*dict(model.named_parameters()).values(), inputs).sum().backward()
         self.assertIsNotNone(model.linear.weight.grad)
 
+    def test_activation_checkpointing_with_meta_preservation(self):
+        """Test that activation checkpointing works correctly with node meta preservation
+
+        This test verifies that the boxed_nop_preserve_node_meta function correctly
+        preserves metadata during initial trace when using HOPs like activation checkpointing,
+        and that the activation checkpointing still functions when torch.compile is applied later.
+        """
+        from torch.utils.checkpoint import checkpoint
+        from torch._functorch._aot_autograd.graph_capture_wrappers import set_partitioner_tag
+
+        class CheckpointedModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear1 = nn.Linear(4, 4, bias=False)
+                self.linear2 = nn.Linear(4, 4, bias=False)
+
+            def _checkpointed_layers(self, x):
+                x = torch.relu(self.linear1(x))
+                return x
+
+            def forward(self, x):
+                x = checkpoint(self._checkpointed_layers, x, use_reentrant=False)
+                #x = self._checkpointed_layers(x)
+                x = torch.relu(self.linear2(x))
+                return x
+
+        from torch._subclasses.fake_tensor import FakeTensorMode
+        with FakeTensorMode():
+            model = CheckpointedModule()
+            inputs = (torch.randn(2, 4, requires_grad=True),)
```

Comment:
fakeification only needed because i forced us to trace through saved tensor hooks

Reply:
TODO: Should error if new saved tensor hook gets applied in trace region